src/network.lisp

```diff
@@ -23,36 +23,37 @@
   (:report (lambda (condition stream)
              (format stream "Request error: ~A" (what condition)))))
 
+(defun process-options (options return-alist-p)
+  "Process request options, returning alist if required"
+  (let* ((options (alexandria:remove-from-plist options :timeout :streamp))
+	 (sanitized-options (loop for (key value) on options by #'cddr
+				  when value
+				    collect (kebab:to-snake-case key)
+				    and collect value)))
+    (if return-alist-p
+	(alexandria:plist-alist sanitized-options)
+	sanitized-options)))
 
 (defun make-request (bot name &rest options &key (streamp nil) (timeout 3) &allow-other-keys)
-  "Perform HTTP request to 'name API method with 'options JSON-encoded object."
+  "Perform HTTP request to 'name API method with 'options JSON-encoded object or multi-part form-data."
   (declare (ignore streamp))
 
   (let ((url (concatenate 'string (get-endpoint bot) name)))
     (log:debug "Posting data to"
                (obfuscate url)
                options)
     (let* ((max-timeout (* timeout 10))
-           (processed-options (loop for (key value)
-                                      on (alexandria:remove-from-plist options :timeout :streamp)
-                                        by #'cddr
-                                    when value
-                                      collect (kebab:to-snake-case key)
-                                      and
-                                        collect value))
+	   (multipart-required (find-if #'pathnamep options))
```

Comment:
Probably it will be better to use `(some #'pathnamep options) here, because we don't need a pathname as the result:

```
CL-USER> (some #'pathnamep (list 1 2 #P"foo" #P"bar" 5 6))
T
CL-USER> (find-if #'pathnamep (list 1 2 #P"foo" #P"bar" 5 6))
#P"foo"
```